lib/src/palantirJavaFormat/java/com/diffplug/spotless/glue/pjf/PalantirJavaFormatFormatterFormatSourceAndFixImportsAndDeclarationsFunc.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022-2025 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.glue.pjf;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import com.palantir.javaformat.java.Formatter;
+import com.palantir.javaformat.java.ImportOrderer;
+import com.palantir.javaformat.java.JavaFormatterOptions;
+import com.palantir.javaformat.java.RemoveUnusedImports;
+
+import com.diffplug.spotless.FormatterFunc;
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.Lint;
+
+public class PalantirJavaFormatFormatterFormatSourceAndFixImportsAndDeclarationsFunc implements FormatterFunc, FormatterStep {
+
+	private final Formatter formatter;
+
+	private final JavaFormatterOptions.Style formatterStyle;
+
+	/**
+	 * Creates a new formatter func that formats code via Palantir.
+	 * @param style The style to use for formatting.
+	 * @param formatJavadoc Whether to format Java docs. Requires at least Palantir 2.36.0 or later, otherwise the
+	 * constructor will throw.
+	 */
+	public PalantirJavaFormatFormatterFormatSourceAndFixImportsAndDeclarationsFunc(String style, boolean formatJavadoc) {
+		this.formatterStyle = JavaFormatterOptions.Style.valueOf(style);
+		JavaFormatterOptions.Builder builder = JavaFormatterOptions.builder();
+		builder.style(formatterStyle);
+		if (formatJavadoc) {
+			applyFormatJavadoc(builder);
+		}
+		formatter = Formatter.createFormatter(builder.build());
+	}
+
+	@Override
+	public String apply(String input) throws Exception {
+		return formatter.formatSourceAndFixImportsAndDeclarations(input);
```

Comment:
Cannot resolve method 'formatSourceAndFixImportsAndDeclarations' in 'Formatter'

- https://github.com/palantir/palantir-java-format/pull/1342